back/spec/models/initiative_spec.rb

```diff
@@ -218,22 +218,30 @@
       expect(initiative.reload.cosponsors).to match_array [cosponsor1, cosponsor2]
     end
 
-    it 'removes cosponsor when given array of user IDs excluding ID of existing cosponsor' do
+    it 'removes cosponsors_initiative when given array of user IDs excluding ID of existing cosponsor' do
       cosponsor2 = create(:user)
       initiative.update!(cosponsor_ids: [cosponsor2.id])
 
       expect(initiative.reload.cosponsors).to match_array [cosponsor2]
     end
 
-    it 'can add and remove cosponsors at the same time' do
+    it 'removes cosponsors_initiative even when an associated notifcation exists' do
+      cosponsor2 = create(:user)
+      create(:invitation_to_cosponsor_initiative, cosponsors_initiative: cosponsors_initiative)
+      initiative.update!(cosponsor_ids: [cosponsor2.id])
+
+      expect(initiative.reload.cosponsors).to match_array [cosponsor2]
+    end
```

Comment:
New regression test, to ensure we can destroy `cosponsors_initiatives` even when associated `notifications` exist.